cmd/connect.go

```diff
@@ -73,30 +91,121 @@ of the workload cluster (e.g., 've5v6' for 'enigma-ve5v6').
 			teleportContextToUse = "teleport.giantswarm.io-" + fullWorkloadClusterName
 		}
 
-		// Note: The original `tsh` output is now manually printed above.
-		// The TUI will later capture and display `tsh` output for *new* connections made via the UI.
-
 		fmt.Printf("Current Kubernetes context set to: %s\n", teleportContextToUse)
-
-		// --- Print Initial Setup Info (can be displayed in TUI later) ---
 		fmt.Println("--------------------------")
-		fmt.Println("Setup complete. Initializing TUI...")
-		// Provider determination can be moved into the TUI model or passed if needed
-		// For now, keeping it simple and focusing on port-forward display
 
-		// Initialize the TUI model
-		tuiModel := tui.InitialModel(managementCluster, fullWorkloadClusterName, teleportContextToUse)
-		p := tea.NewProgram(tuiModel, tea.WithAltScreen()) // tea.WithOutput(os.Stderr) can be useful for debugging
+		if noTUI {
+			fmt.Println("Skipping TUI. Setting up port forwarding in the background...")
+			// Placeholder for non-TUI port forwarding logic
+			// This will involve calling a modified version of port forwarding setup
 
-		if _, err := p.Run(); err != nil {
-			fmt.Fprintf(os.Stderr, "Error running TUI: %v\n", err)
-			return err
-		}
+			// Get port forwarding configurations
+			configs := getPortForwardConfigs(managementCluster, fullWorkloadClusterName, teleportContextToUse)
+			if len(configs) == 0 {
+				fmt.Println("No port forwarding configurations found. Exiting.")
+				return nil
+			}
+
+			var wg sync.WaitGroup
+			stopChannels := make([]chan struct{}, 0)
+			allStopChan := make(chan struct{}) // Single channel to signal all goroutines
+
+			for _, pfConfig := range configs {
+				wg.Add(1)
+				// Use a local copy of pfConfig for the goroutine
+				config := pfConfig
+				go func() {
+					defer wg.Done()
+					fmt.Printf("Attempting to start port-forward for %s on %s to %s:%s (context: %s)...\n",
+						config.label, config.service, config.localPort, config.remotePort, config.kubeContext)
+
+					// Simple console logger for updates
+					sendUpdateFunc := func(status, outputLog string, isError, isReady bool) {
```

Comment:
Consider adding context handling for port forwarding goroutines to prevent potential resource leaks. Also add port availability checks before binding:

```go
ctx, cancel := context.WithCancel(context.Background())
defer cancel()

// Check port availability
if err := utils.CheckPortAvailable(config.localPort); err != nil {
    return fmt.Errorf("port %s is not available: %w", config.localPort, err)
}

// Use context in port forwarding
individualStopChan, initialStatus, initialErr := utils.StartPortForwardClientGo(
    ctx,
    config.kubeContext,
    config.namespace,
    config.service,
    portSpec,
    config.label,
    sendUpdateFunc,
)
```